src/engraving/libmscore/excerpt.cpp

```diff
@@ -667,6 +667,12 @@ void Excerpt::cloneSpanner(Spanner* s, Score* score, track_idx_t dstTrack, track
         if (!ns->endElement()) {
             LOGD("clone Slur: no end element");
         }
+    } else if (ns->isTrill()) {
+        EngravingItem* startElement = (EngravingItem*)s->startElement()->findLinkedInScore(score);
+        if (startElement && startElement->isChord()) {
+            ns->setStartElement(startElement);
+            startElement->startingSpanners().push_back(ns);
```

Comment:
Do we really need these startingSpanners/endingSpanners in EngravingItem? They do not make much sense for a lot of engraving elements (but each of them has these collections; therefore, they consume a decent amount of memory). Also, they are pretty vague in terms of memory management: if any spanner is deleted, an invalid pointer will remain in these collections

Reply:
I did it originally for performance reasons: there are many situations where an item needs know that a spanner is starting or ending on it, and searching through the SpannerMap every time became too expensive. Keeping these collections makes it more immediate (and the collection is updated from Score::removeSpanner when one is deleted, which seems to work reliably so far). You're definitely right though that it isn't needed for all EngravingItems, I can probably restrict it to Chord, or ChordRest. I'll work on it now

Reply:
@RomanPudashkin  ok, done! I've moved `startingSpanners` and `endingSpanners` to the Chord class. And I've made them std::set instead of std::vector, which makes more sense. Also, I've enclosed them better and avoided passing them by reference outside the class.

Reply:
It looks better, but still... Can we move these sets to LayoutContext? They look like they should be there

Reply:
At the moment, I wouldn't know how to do that. The bottom line is that, for every chord, we need to know about spanners starting or ending on it, because several operations depend on this info. When I first did it, I was just looking into the SpannerMap everytime I needed this info, but it was too expensive, so I had to create and maintain these collections for each chord instead. Now, if we move these sets to the LayoutState, a) they would be created/destroyed at every layout b) we would still need one for each chord, which would mean creating some sort of map like `std::map<Chord*, std::set<Spanner*>> startingSpanners` or something like that, so I'm afraid I'd be going back to the same performance problem I had originally. I agree with you that I don't like maintaining these sets for each chord, but I don't see an immediate solution. I would try to tackle this as a separate job